tests/pytests/test_issues.py

```diff
@@ -1209,6 +1209,7 @@ def test_mod_7495(env: Env):
 @skip(cluster=True)
 def test_mod_8142(env:Env):
   env.expect('FT.CREATE', 'idx', 'SCHEMA', 't', 'TEXT').ok()
+  env.expect('FT.CREATE', 'idxOptimizedOptional', 'INDEXALL', 'ENABLE', 'SCHEMA', 't', 'TEXT').ok()
```

Comment:
We also test the `NOT` iterator here, which also benefits from the `INDEXALL` optimization. Consider renaming the index:
```suggestion
  env.expect('FT.CREATE', 'idxOptimized', 'INDEXALL', 'ENABLE', 'SCHEMA', 't', 'TEXT').ok()
```
```suggestion
  env.expect('FT.CREATE', 'idxIndexAll', 'INDEXALL', 'ENABLE', 'SCHEMA', 't', 'TEXT').ok()
```